CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# -------------------- Project Information ------------------- #
+
+cmake_minimum_required(VERSION 3.25.1)
+
+project(stream)
+
+# ------------------- Environment Settings ------------------- #
+
+if (APPLE)
+    # Need llvm installed.
+    set(CMAKE_CXX_COMPILER /usr/local/opt/llvm/bin/clang++)
+    set(CMAKE_C_COMPILER /usr/local/opt/llvm/bin/clang)
+    # Link llvm libraries.
+    link_directories(/usr/local/opt/llvm/lib)
```

Comment:
I don't think hardcoding absolute paths is a good idea at all, even more so if there are homebrew paths.